src/avalanchemq/queue/durable_queue.cr

```diff
@@ -228,6 +220,24 @@ module AvalancheMQ
       @log.error { "Could not restore index: #{ex.inspect}" }
     end
 
+    # Reads all SPs in the index, truncate if there's only 0's left
+    # As we preallocate files, they are not truncated on ungraceful shutdown
+    # and thus we allocate too large memory structures in restore_index.
+    # TODO: optimize using SEEK_HOLE or compare allocated blocks * block size vs file size
+    private def truncate_sparse_file(path)
+      File.open(path, "W") do |f|
+        f.buffer_size = Config.instance.file_buffer_size
+        f.advise(File::Advice::Sequential)
+        sp = SegmentPosition.from_io file
```

Comment:
Dont you need to loop here to go through the whole file? 

and the variable is called `file` here but `f` in the block

Reply:
thanks